api/src/main/java/jakarta/data/BasicRestrictionRecord.java

```diff
@@ -36,4 +36,33 @@ record BasicRestrictionRecord<T>(
     BasicRestrictionRecord(String field, Operator comparison, Object value) {
         this(field, false, comparison, value);
     }
+
+    @Override
+    public Restriction<T> negate() {
+        boolean newNegation = isNegated;
+        Operator newComparison;
+        switch (comparison) {
+            case GREATER_THAN:
+                newComparison = Operator.LESS_THAN_EQUAL;
+                break;
+            case GREATER_THAN_EQUAL:
+                newComparison = Operator.LESS_THAN;
+                break;
+            case LESS_THAN:
+                newComparison = Operator.GREATER_THAN_EQUAL;
+                break;
+            case LESS_THAN_EQUAL:
+                newComparison = Operator.GREATER_THAN;
+                break;
+            default:
+                newComparison = comparison;
+                newNegation = !isNegated;
```

Comment:
Actually I think it would make more sense to just add `NOT_IN` and `NOT_LIKE` operators (they exist in JDQL), and move all this logic to a `negated()` method on `Operator`.

Reply:
Good idea. That makes it possible for all operations to be negated and then makes sense for negation to move to Operator as you suggested. I added this in commit https://github.com/jakartaee/data/pull/905/commits/c480490190e05b90954166d48eb66d2d970280aa